src/prefect/runner/storage.py

```diff
@@ -276,6 +279,137 @@ def to_pull_step(self) -> Dict:
+        self._logger = get_logger("runner.storage.remote-storage")
+        self._storage_base_path = Path.cwd()
+        self._pull_interval = pull_interval
+
+    @staticmethod
+    def get_required_package_for_scheme(scheme: str) -> Optional[str]:
+        # attempt to discover the package name for the given scheme
+        # from fsspec's registry
+        known_implementation = fsspec.registry.get(scheme)
+        if known_implementation:
+            return known_implementation.__module__.split(".")[0]
+        # if we don't know the implementation, try to guess it for some
+        # common schemes
+        elif scheme == "s3":
+            return "s3fs"
+        elif scheme == "gs" or scheme == "gcs":
+            return "gcsfs"
+        elif scheme == "abfs" or scheme == "az":
+            return "adlfs"
+        else:
+            None
+
+    @property
+    def filesystem(self) -> fsspec.AbstractFileSystem:
+        scheme, _, _, _, _ = urlsplit(self._url)
+
+        def replace_blocks_with_values(obj: Any) -> Any:
+            if isinstance(obj, Block):
+                if hasattr(obj, "get"):
+                    return obj.get()
+                if hasattr(obj, "value"):
+                    return obj.value
+                else:
+                    return obj.dict()
+            return obj
+
+        settings_with_block_values = visit_collection(
+            self._settings, replace_blocks_with_values, return_data=True
+        )
+
+        return fsspec.filesystem(scheme, **settings_with_block_values)
+
+    def set_base_path(self, path: Path):
+        self._storage_base_path = path
+
+    @property
+    def pull_interval(self) -> Optional[int]:
+        """
+        The interval at which contents from remote storage should be pulled to
+        local storage. If None, remote storage will perform a one-time sync.
+        """
+        return self._pull_interval
+
+    @property
+    def destination(self) -> Path:
+        """
+        The local file path to pull contents from remote storage to.
+        """
+        _, netloc, urlpath, _, _ = urlsplit(self._url)
+        return self._storage_base_path / Path(netloc) / Path(urlpath.lstrip("/"))
+
+    async def pull_code(self):
+        """
+        Pulls contents from remote storage to the local filesystem.
+        """
+        self._logger.debug(
+            "Pulling contents from remote storage '%s' to '%s'...",
+            self._url,
+            self.destination,
+        )
+        _, netloc, urlpath, _, _ = urlsplit(self._url)
+
+        if not self.destination.exists():
+            self.destination.mkdir(parents=True, exist_ok=True)
+
+        remote_path = str(Path(netloc) / Path(urlpath.lstrip("/"))) + "/"
+
+        await from_async.wait_for_call_in_new_thread(
+            create_call(
+                self.filesystem.get,
+                remote_path,
+                str(self.destination),
+                recursive=True,
+            )
+        )
```

Comment:
Does it make sense to put this in a `try` and `except` block with like `f"Failed to pull from {remote_path} due to e"`